packages/babel-helpers/src/helpers.js

```diff
@@ -270,6 +270,151 @@ helpers.createClass = template(`
   })()
 `);
 
+//NOTE: convention is 'descriptor' is used for element descriptors, while 'propertyDescriptor' is used for
+//property descriptor in all helpers related to decorators-2
+helpers.decorate = template(`
+  (function (constructor, undecorated, memberDecorators, heritage) {
+    const prototype = constructor.prototype;
+    let finishers = [];
+    const elementDescriptors = new Map(); // elementDescriptors is meant to be an array, so this will be converted later
+
+    for (const [key, isStatic] of undecorated) {
+      const target = isStatic ? constructor : prototype;
+      const propertyDescriptor = Object.getOwnPropertyDescriptor(target, key);
+      elementDescriptors.set(
+        [key, isStatic], 
+        babelHelpers.makeElementDescriptor(
+          "property",
+          key,
+          isStatic,
+          propertyDescriptor,
+        )
+      );
+    }
+
+    // decorate and store in elementDescriptors
+    for (const [key, decorators, isStatic] of memberDecorators) {
+      const target = isStatic ? constructor : prototype;
+      const propertyDescriptor =
+        elementDescriptors.has([key, isStatic]) && elementDescriptors.get([key, isStatic]).descriptor 
```

Comment:
This has call will always return false, since ES6 maps are by identity. Since [1, 2] !== [1, 2], calling Map.prototype.has on a fresh array will never do what you're hoping for.

Instead, you could maintain two Sets of keys, or two objects with keys and values, for the static and non-static halves.